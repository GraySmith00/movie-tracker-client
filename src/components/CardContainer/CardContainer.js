import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const CardContainer = ({ movies, category }) => {
  const displayCards = movies[category].map(movie => (
    <MovieCard key={movie.id} movie={movie} />
  ));

  return <div>{displayCards}</div>;
};

CardContainer.propTypes = {
  movies: PropTypes.object.isRequired,
  category: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  movies: state.movies
});

export default connect(mapStateToProps)(CardContainer);
