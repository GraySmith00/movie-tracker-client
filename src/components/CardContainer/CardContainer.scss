.card-container {
  margin: 2rem auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);

  grid-gap: 2em;
  width: 90%;
}
