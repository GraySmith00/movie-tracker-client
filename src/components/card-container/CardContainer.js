import React from "react";
import MovieCard from "../movie-card/MovieCard";

const CardContainer = ({ data, path }) => {
  const cardContainer = data.map((movieObj, index) => (
    <MovieCard key={`Container-${index}`} movieObj={movieObj} path={path} />
  ));
  return <div> {cardContainer}</div>;
};

export default CardContainer;
