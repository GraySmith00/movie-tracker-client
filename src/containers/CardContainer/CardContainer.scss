.card-container {
  margin: 2rem auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);

  grid-gap: 2em;
  width: 90%;
}

@media screen and (max-width: 1340px) {
  .card-container {
    grid-gap: 0.8em;
  }
}

@media screen and (max-width: 1280px) {
  .card-container {
    grid-gap: 2em;
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (max-width: 1060px) {
  .card-container {
    grid-gap: 0.8em;
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (max-width: 1000px) {
  .card-container {
    grid-gap: 2em;
    grid-template-columns: repeat(2, 1fr);
  }
}

@media screen and (max-width: 540px) {
  .card-container {
    grid-template-columns: repeat(1, 1fr);
    justify-content: center;
    padding: 0 25%;
  }
}
