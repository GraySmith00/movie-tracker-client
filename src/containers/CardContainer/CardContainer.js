import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './CardContainer.css';

export const CardContainer = ({ movies, category }) => {
  let displayCards;
  if (category === 'favorites') {
    if (movies.favorites.length === 0) {
      displayCards = <p>You have not added anything to your favorites yet!</p>;
    } else {
      displayCards = movies[category].map(movieId => {
        const movie = movies.nowPlaying.find(
          movie => movie.movie_id === movieId
        );
        return <MovieCard key={movie.movie_id} movie={movie} />;
      });
    }
  } else {
    displayCards = movies[category].map(movie => {
      return <MovieCard key={movie.movie_id} movie={movie} />;
    });
  }

  return <div className="card-container">{displayCards}</div>;
};

CardContainer.propTypes = {
  movies: PropTypes.object.isRequired,
  category: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  movies: state.movies
});

export default connect(mapStateToProps)(CardContainer);
