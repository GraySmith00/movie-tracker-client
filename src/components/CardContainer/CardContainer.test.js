import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
