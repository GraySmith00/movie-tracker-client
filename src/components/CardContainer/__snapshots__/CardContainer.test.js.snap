// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `undefined`;
